test/test_sparse_csr.py

```diff
@@ -1,70 +1,87 @@
-
-        for f in fills:
-            dense[f] = 0
-        dense = torch.from_numpy(dense.reshape(shape))
-
-        return dense.to_sparse_csr()
-
-    def setUp(self):
-        # These parameters control the various ways we can run the test.
-        # We will subclass and override this method to implement CUDA
-        # tests
-        self.is_cuda = False
-        self.device = 'cpu'
-        self.index_tensor = lambda *args: torch.tensor(*args, dtype=torch.int32)
-        self.value_tensor = lambda *args: torch.tensor(*args, dtype=torch.double)
-
-    def test_csr_layout(self):
+
+    @onlyCPU
+    def test_csr_layout(self, _):
         self.assertEqual(str(torch.sparse_csr), 'torch.sparse_csr')
         self.assertEqual(type(torch.sparse_csr), torch.layout)
 
-    def test_sparse_csr_constructor_shape_inference(self):
+    @onlyCPU
+    @dtypes(torch.double)
+    def test_sparse_csr_constructor_shape_inference(self, device, dtype):
         crow_indices = [0, 2, 4]
         col_indices = [0, 1, 0, 1]
         values = [1, 2, 3, 4]
         sparse = torch.sparse_csr_tensor(torch.tensor(crow_indices, dtype=torch.int64),
                                          torch.tensor(col_indices, dtype=torch.int64),
-                                         torch.tensor(values), dtype=torch.double)
+                                         torch.tensor(values), dtype=dtype, device=device)
         self.assertEqual(torch.tensor(crow_indices, dtype=torch.int64), sparse.crow_indices())
         self.assertEqual((len(crow_indices) - 1, max(col_indices) + 1), sparse.shape)
 
-    def test_sparse_csr_constructor(self):
+    @onlyCPU
+    @dtypes(*torch.testing.get_all_dtypes(include_bool=False, include_half=False,
+                                          include_bfloat16=False, include_complex=False))
+    def test_sparse_csr_constructor(self, device, dtype):
         crow_indices = [0, 2, 4]
         col_indices = [0, 1, 0, 1]
         values = [1, 2, 3, 4]
-
-        sparse = torch.sparse_csr_tensor(torch.tensor(crow_indices, dtype=torch.int32),
-                                         torch.tensor(col_indices, dtype=torch.int32),
-                                         torch.tensor(values), size=(2, 10), dtype=torch.float)
-
-        self.assertEqual((2, 10), sparse.shape)
-        self.assertEqual(torch.tensor(crow_indices, dtype=torch.int32), sparse.crow_indices())
-
-    def test_sparse_csr_print(self):
+        for index_dtype in [torch.int32, torch.int64]:
+            sparse = torch.sparse_csr_tensor(torch.tensor(crow_indices, dtype=index_dtype),
+                                             torch.tensor(col_indices, dtype=index_dtype),
+                                             torch.tensor(values),
+                                             size=(2, 10),
+                                             dtype=dtype,
+                                             device=device)
+            self.assertEqual((2, 10), sparse.shape)
```

Comment:
Could you also add dtype/device checks:
```
self.assertEqual(dtype, sparse.dtype)
self.assertEqual(device, sparse.device)
```
to be sure that we get what we request.